### DISCUSSION TITLE
[2023-03-29] Study Topic - GPIO MPRJ IO

### DISCUSSION THREAD

**Discussion by boledulab** [bol-edu/caravel-soc#26] [General]
[GPIO_MPRJ_IO by Willy.pdf](https://github.com/bol-edu/caravel-soc/files/11109186/GPIO_MPRJ_IO.by.Willy.pdf)

**Comment by TonyHo722** [comment]
video in 全端 IC 設計工程師伴學松 第七次 1:01:24 ~ 1:38:11
Architecture Introduction
1:01:24 One pin GPIO
1:13:15 mprj-io

https://www.youtube.com/watch?v=C-bZV4xIqs8

**Comment by TonyHo722** [comment]
in below link, 雖然上面寫到使用者可改變caravel chip default setting when power on. 但是若是使用 [skywater MPW](https://www.skywatertechnology.com/technology-and-design-enablement/mpw-programs/) 的流程，user_defines.v 應該也是不能改的。 除非是連同 caravel management core 都可以更改的 tape-out flow，user才能自行定義caravel chip default setting when power on.
https://github.com/bol-edu/caravel-soc/blob/cfbb1d4f13311d595f0f06f78947e16302339e6c/rtl/header/user_defines.v#L38-L59

**Comment by TonyHo722** [comment]
NO_TOP_LEVEL_BUFFERING should reference this code
https://github.com/bol-edu/fsic_asic/blob/a4562f057d31eeaef165d9404c47a9e3c389067d/dsn/rtl/caravel.v#L295-L322
